assets/components/emailFormField/emailFormField.jsx

```diff
@@ -41,19 +34,32 @@ const EmailFormField = (props: PropTypes) => {
     modifierClass.push('error');
   }
 
+  // Copied from
+  // https://github.com/playframework/playframework/blob/38abd1ca6d17237950c82b1483057c5c39929cb4/framework/src/play/
+  // src/main/scala/play/api/data/validation/Validation.scala#L80
+  // but with minor modification (last * becomes +) to enforce at least one dot in domain.  This is
+  // for compatibility with Stripe
+  const emailRegexPattern =  "[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$";
```

Comment:
Do you still need this despite `type="email"`?

Reply:
(In response to your comment, you can add a `pattern="<regexp>"` to make sure the email address terminates with `<domain>.<tld>` and let the browser do the whole work)